lib/pharos/reboot_command.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Pharos
+  class RebootCommand < Pharos::Command
+    options :filtered_hosts, :yes?
+
+    def execute
+      Dir.chdir(config_yaml.dirname) do
+        filtered_hosts.size == load_config.hosts.size ? reboot_all : reboot_hosts
+      end
+    end
+
+    def reboot_all
+      confirm_yes!(pastel.bright_yellow("==> Do you really want to reboot all hosts in the cluster?"), default: false)
+      reboot_hosts
+    end
+
+    def reboot_hosts
+      start_time = Time.now
+
+      master_hosts = filtered_hosts.select(&:master?)
+      worker_hosts = filtered_hosts.select(&:worker?)
+
+      puts pastel.green("==> Sharpening tools ...")
+      cluster_manager.gather_facts
+
+      unless master_hosts.empty?
+        puts pastel.green("==> Rebooting #{master_hosts.size} master node#{'s' if master_hosts.size > 1} ...")
+        cluster_manager.apply_reboot_hosts(master_hosts, parallel: false)
+      end
+
+      puts pastel.green("==> Resharpening tools ...")
```

Comment:
After rebooting masters, it seemed logical to regather facts, but now I realize it may never reach this point if the master reboot failed to bring up kubelet again. So, this could be skipped I think.